### PR TITLE
Update deps 2021-07-26

### PR DESCRIPTION
**Pull request by Jimbo4350**

### CODE REVIEW

File: cardano-api/src/Cardano/Api.hs [hunk 1]
```diff
@@ -565,6 +565,8 @@ module Cardano.Api (
     -- ** Conversions
     --TODO: arrange not to export these
     toNetworkMagic,
+    fromAlonzoRdmrPtr,
```
Comment (dcoutts) [inline cardano-api/src/Cardano/Api.hs:568]: No, not to be exported from the main API. From the `.Shelley` module at most.

File: cardano-api/src/Cardano/Api.hs [hunk 2]
```diff
@@ -565,6 +565,8 @@ module Cardano.Api (
...
+    renderScriptWitnessIndex,
```
Comment (dcoutts) [inline cardano-api/src/Cardano/Api.hs:569]: Put this next to the export of `ScriptWitnessIndex` itself. It is not a conversion.

File: cardano-api/src/Cardano/Api/Fees.hs [hunk 3]
```diff
@@ -116,12 +116,22 @@ transactionFee txFeeFixed txFeePerByte tx =
...
+  obtainHasField
+    :: ShelleyLedgerEra era ~ ledgerera
+    => ShelleyBasedEra era
+    -> ( HasField "txsize" (Ledger.Tx (ShelleyLedgerEra era)) Integer
+        => a)
+    -> a
+  obtainHasField ShelleyBasedEraShelley f = f
+  obtainHasField ShelleyBasedEraAllegra f = f
+  obtainHasField ShelleyBasedEraMary    f = f
+  obtainHasField ShelleyBasedEraAlonzo  f = f
```
Comment (dcoutts) [inline cardano-api/src/Cardano/Api/Fees.hs:134]: Note that we should really deprecate `transactionFee`. It is wrong for Alonzo. It should be replaced by `evaluateTransactionFee`.

File: cardano-api/src/Cardano/Api/Tx.hs [hunk 4]
```diff
@@ -409,52 +410,55 @@ getTxBody (ByronTx Byron.ATxAux { Byron.aTaTx = txbody }) =
...
+                                   _bootWits
+                                   txscripts
+                                   txdats
+                                   redeemers,
+                    Alonzo.auxiliaryData = auxiliaryData
+                  } =
+    ShelleyTxBody era txbody
+                  (Map.elems txscripts)
+                  (TxBodyScriptData scriptDataInEra txdats redeemers)
+                  (strictMaybeToMaybe auxiliaryData)
```
Comment (dcoutts) [inline cardano-api/src/Cardano/Api/Tx.hs:461]: There's a lot of changes here, both specifically here and in the rest of the module. It's unclear what are the forced changes due to the update and what is reformatting. Could we try to split those up? i.e. keep _just_ the necessary changes, and if we want to do reformatting or other refactoring, have those in separate commits, either before or after the dep update.

**Review by dcoutts — CHANGES_REQUESTED** [review]

**Review by dcoutts — APPROVED** [review]

**Comment by dcoutts** [comment]
bors merge

**Comment by dcoutts** [comment]
Hmm, the plutus example CI instance failing the build is really odd. It's as if it is not using the right ledger commit. The final patch is exactly about removing the type applications that are no longer necessary, but those builds are failing as if they are still necessary.

**Review by coot — APPROVED** [review]

**Comment by Jimbo4350** [comment]
bors merge

**Pull request merged**